src/toolkits/drawing_classifier/dc_data_iterator.cpp

```diff
@@ -192,41 +195,33 @@ data_iterator::batch simple_data_iterator::next_batch(size_t batch_size) {
   }
 
   if (target_index_ >= 0) {
-    DASSERT_EQ(real_batch_size, batch_targets.size());
+    batch_targets.resize(batch_size, 0.f);
```

Comment:
not sure why you don't want to use `real_batch_size`, which is quite critical.

Reply:
I am making the batch_targets to be of batch_size so assertion makes no sense to me.